[mchehab-zbar] build failure

Package: mchehab-zbar[core]:x64-windows -> 0.23.90#2

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.34.31937.0
-    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 17030920a 2023-01-31 (13 hours ago)

**To Reproduce**

`vcpkg install mchehab-zbar:x64-windows`

**Failure logs**

```
-- Using cached mchehab-zbar-0.23.90.tar.gz.
-- Cleaning sources at E:/vcpkg/buildtrees/mchehab-zbar/src/0.23.90-847dc98386.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/mchehab-zbar-0.23.90.tar.gz
-- Applying patch c99.patch
-- Applying patch issue219.patch
-- Applying patch windows.patch
-- Applying patch x64.patch
-- Using source at E:/vcpkg/buildtrees/mchehab-zbar/src/0.23.90-847dc98386.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows
-- Using cached msys-gzip-1.11-1-x86_64.pkg.tar.zst.
-- Using cached msys-bash-5.1.008-1-x86_64.pkg.tar.zst.
-- Using cached msys-autoconf-2.71-3-any.pkg.tar.zst.
-- Using cached msys-diffutils-3.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-binutils-2.37-5-x86_64.pkg.tar.zst.
-- Using cached msys-libtool-2.4.7-3-x86_64.pkg.tar.zst.
-- Using cached msys-file-5.41-2-x86_64.pkg.tar.zst.
-- Using cached msys-zlib-1.2.11-1-x86_64.pkg.tar.xz.
-- Using cached msys-libbz2-1.0.8-3-x86_64.pkg.tar.zst.
-- Using cached msys-coreutils-8.32-2-x86_64.pkg.tar.zst.
-- Using cached msys-grep-3.0-2-x86_64.pkg.tar.xz.
-- Using cached msys-sed-4.8-2-x86_64.pkg.tar.zst.
-- Using cached msys-libpcre-8.45-1-x86_64.pkg.tar.zst.
-- Using cached msys-m4-1.4.19-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake-wrapper-11-4-any.pkg.tar.zst.
-- Using cached msys-gawk-5.1.0-2-x86_64.pkg.tar.zst.
-- Using cached msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-gmp-6.2.1-1-x86_64.pkg.tar.zst.
-- Using cached msys-libreadline-8.1.001-1-x86_64.pkg.tar.zst.
-- Using cached msys-ncurses-6.2-2-x86_64.pkg.tar.zst.
-- Using cached msys-automake1.16-1.16.3-3-any.pkg.tar.zst.
-- Using cached msys-perl-5.32.1-2-x86_64.pkg.tar.zst.
-- Using cached msys-libcrypt-2.1-3-x86_64.pkg.tar.zst.
-- Using cached msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst.
-- Using cached msys-make-4.3-3-x86_64.pkg.tar.zst.
-- Using cached msys-findutils-4.8.0-1-x86_64.pkg.tar.zst.
-- Using cached msys-libintl-0.21-1-x86_64.pkg.tar.zst.
-- Using cached msys-libiconv-1.16-2-x86_64.pkg.tar.zst.
-- Using cached msys-gcc-libs-11.2.0-3-x86_64.pkg.tar.zst.
-- Using cached msys-msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst.
-- Using cached msys-which-2.21-4-x86_64.pkg.tar.zst.
-- Using msys root at E:/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5
-- Generating configure for x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: E:/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5/usr/bin/bash.exe --noprofile --norc --debug -c "autoreconf -vfi"
    Working Directory: E:/vcpkg/buildtrees/mchehab-zbar/src/0.23.90-847dc98386.clean/
    Error code: 1
    See logs for more information:
      E:\vcpkg\buildtrees\mchehab-zbar\autoconf-x64-windows-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:604 (vcpkg_execute_required_process)
  ports/mchehab-zbar/portfile.cmake:21 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)



```
<details><summary>E:\vcpkg\buildtrees\mchehab-zbar\autoconf-x64-windows-err.log</summary>

```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
Can't exec "autopoint": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 293.
autoreconf: error: autopoint failed with exit status: 1
```
</details>

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -17,8 +17,20 @@ if("nls" IN_LIST FEATURES)
 else()
     set(ENV{AUTOPOINT} true) # true, the program
     file(TOUCH "${SOURCE_PATH}/po/Makefile.in.in")
-    file(WRITE "${SOURCE_PATH}/config/iconv.m4" "AC_DEFUN([AM_ICONV],[])\n")
-    file(WRITE "${SOURCE_PATH}/config/gettext.m4" "AC_DEFUN([AM_GNU_GETTEXT],[])\nAC_DEFUN([AM_GNU_GETTEXT_VERSION],[])\nAC_DEFUN([AM_GNU_GETTEXT_REQUIRE_VERSION],[])\n")
+    # Get build-time m4 files from gettext
+    set(gettext_version 0.21.1)
+    vcpkg_download_distfile(gettext_archive
+        URLS "https://ftp.gnu.org/pub/gnu/gettext/gettext-${gettext_version}.tar.gz"
+             "https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/gettext/gettext-${gettext_version}.tar.gz"
+        FILENAME "gettext-${gettext_version}.tar.gz"
+        SHA512 ccd43a43fab3c90ed99b3e27628c9aeb7186398153b137a4997f8c7ddfd9729b0ba9d15348567e5206af50ac027673d2b8a3415bb3fc65f87ad778f85dc03a05
+    )
+    file(ARCHIVE_EXTRACT INPUT "${gettext_archive}"
+        DESTINATION "${SOURCE_PATH}/gettext-for-m4"
+        PATTERNS "*/gettext-runtime/m4/gettext.m4" "*/gettext-runtime/m4/iconv.m4"
+    )
+    file(GLOB_RECURSE m4_files "${SOURCE_PATH}/gettext-for-m4/*/*.m4")
+    file(INSTALL ${m4_files} DESTINATION "${SOURCE_PATH}/config")
     vcpkg_list(APPEND options "--disable-nls")
 endif()
 

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -15,6 +15,8 @@ vcpkg_list(SET options)
 if("nls" IN_LIST FEATURES)
     vcpkg_list(APPEND options "--enable-nls")
 else()
+    set(ENV{AUTOPOINT} true) # true, the program
+    file(TOUCH "${SOURCE_PATH}/po/Makefile.in.in")
     vcpkg_list(APPEND options "--disable-nls")
 endif()
 

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -17,6 +17,8 @@ if("nls" IN_LIST FEATURES)
 else()
     set(ENV{AUTOPOINT} true) # true, the program
     file(TOUCH "${SOURCE_PATH}/po/Makefile.in.in")
+    file(WRITE "${SOURCE_PATH}/config/iconv.m4" "AC_DEFUN([AM_ICONV],[])\n")
+    file(WRITE "${SOURCE_PATH}/config/gettext.m4" "AC_DEFUN([AM_GNU_GETTEXT],[])\nAC_DEFUN([AM_GNU_GETTEXT_VERSION],[])\nAC_DEFUN([AM_GNU_GETTEXT_REQUIRE_VERSION],[])\n")
     vcpkg_list(APPEND options "--disable-nls")
 endif()
 

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -15,6 +15,7 @@ vcpkg_list(SET options)
 if("nls" IN_LIST FEATURES)
     vcpkg_list(APPEND options "--enable-nls")
 else()
+    vcpkg_list(APPEND options "--disable-nls")
     set(ENV{AUTOPOINT} true) # true, the program
     file(TOUCH "${SOURCE_PATH}/po/Makefile.in.in")
     # Get missing build-time m4 files from gettext source
@@ -29,6 +30,9 @@ else()
         DESTINATION "${SOURCE_PATH}/gettext-autoconf"
         PATTERNS "*/gettext-runtime/m4/gettext.m4"
                  "*/gettext-runtime/m4/iconv.m4"
+                 "*/gettext-runtime/m4/nls.m4"
+                 "*/gettext-runtime/m4/po.m4"
+                 "*/gettext-runtime/gnulib-m4/lib-link.m4"
                  "*/gettext-runtime/gnulib-m4/lib-prefix.m4"
     )
     file(GLOB_RECURSE m4_files "${SOURCE_PATH}/gettext-autoconf/*/*.m4")

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -30,8 +30,12 @@ else()
         DESTINATION "${SOURCE_PATH}/gettext-autoconf"
         PATTERNS "*/gettext-runtime/m4/gettext.m4"
                  "*/gettext-runtime/m4/iconv.m4"
+                 "*/gettext-runtime/m4/intlmacosx.m4"
                  "*/gettext-runtime/m4/nls.m4"
                  "*/gettext-runtime/m4/po.m4"
+                 "*/gettext-runtime/m4/progtest.m4"
+                 "*/gettext-runtime/gnulib-m4/host-cpu-c-abi.m4"
+                 "*/gettext-runtime/gnulib-m4/lib-ld.m4"
                  "*/gettext-runtime/gnulib-m4/lib-link.m4"
                  "*/gettext-runtime/gnulib-m4/lib-prefix.m4"
     )

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -17,7 +17,7 @@ if("nls" IN_LIST FEATURES)
 else()
     set(ENV{AUTOPOINT} true) # true, the program
     file(TOUCH "${SOURCE_PATH}/po/Makefile.in.in")
-    # Get build-time m4 files from gettext
+    # Get missing build-time m4 files from gettext source
     set(gettext_version 0.21.1)
     vcpkg_download_distfile(gettext_archive
         URLS "https://ftp.gnu.org/pub/gnu/gettext/gettext-${gettext_version}.tar.gz"
@@ -26,12 +26,13 @@ else()
         SHA512 ccd43a43fab3c90ed99b3e27628c9aeb7186398153b137a4997f8c7ddfd9729b0ba9d15348567e5206af50ac027673d2b8a3415bb3fc65f87ad778f85dc03a05
     )
     file(ARCHIVE_EXTRACT INPUT "${gettext_archive}"
-        DESTINATION "${SOURCE_PATH}/gettext-for-m4"
-        PATTERNS "*/gettext-runtime/m4/gettext.m4" "*/gettext-runtime/m4/iconv.m4"
+        DESTINATION "${SOURCE_PATH}/gettext-autoconf"
+        PATTERNS "*/gettext-runtime/m4/gettext.m4"
+                 "*/gettext-runtime/m4/iconv.m4"
+                 "*/gettext-runtime/gnulib-m4/lib-prefix.m4"
     )
-    file(GLOB_RECURSE m4_files "${SOURCE_PATH}/gettext-for-m4/*/*.m4")
+    file(GLOB_RECURSE m4_files "${SOURCE_PATH}/gettext-autoconf/*/*.m4")
     file(INSTALL ${m4_files} DESTINATION "${SOURCE_PATH}/config")
-    vcpkg_list(APPEND options "--disable-nls")
 endif()
 
 vcpkg_configure_make(

--- a/ports/mchehab-zbar/portfile.cmake
+++ b/ports/mchehab-zbar/portfile.cmake
@@ -46,6 +46,7 @@ endif()
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
     AUTOCONFIG
+    ADD_BIN_TO_PATH # checking for working iconv
     OPTIONS
         ${options}
         --without-dbus

--- a/ports/mchehab-zbar/vcpkg.json
+++ b/ports/mchehab-zbar/vcpkg.json
@@ -1,16 +1,12 @@
 {
   "name": "mchehab-zbar",
   "version": "0.23.90",
-  "port-version": 2,
+  "port-version": 3,
   "description": "ZBar is an open source software suite for reading bar codes from various sources, including webcams. This fork is actively maintained.",
   "homepage": "https://github.com/mchehab/zbar",
   "license": "LGPL-2.1-or-later",
   "supports": "!uwp",
   "dependencies": [
-    {
-      "name": "gettext",
-      "host": true
-    },
     "libiconv"
   ],
   "features": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4906,7 +4906,7 @@
     },
     "mchehab-zbar": {
       "baseline": "0.23.90",
-      "port-version": 2
+      "port-version": 3
     },
     "mcpp": {
       "baseline": "2.7.2.14",

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "78af6695c1ae1889782321149f71f12547f6ea66",
+      "git-tree": "25b923000586733fcae49bf028e7d6be7aea5b1c",
       "version": "0.23.90",
       "port-version": 3
     },

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "3ed6ffc669564d78e83a01eead9a2c2b8e4c8766",
+      "git-tree": "935ceb14fbc62e378361d15466182ae4b88cf37a",
       "version": "0.23.90",
       "port-version": 3
     },

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "692b86ac5546c9b25e2c126653ca1d5f55002d88",
+      "version": "0.23.90",
+      "port-version": 3
+    },
+    {
       "git-tree": "af1f66aef9076cd2e27c941851d1549d1cbffe00",
       "version": "0.23.90",
       "port-version": 2

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "692b86ac5546c9b25e2c126653ca1d5f55002d88",
+      "git-tree": "3ed6ffc669564d78e83a01eead9a2c2b8e4c8766",
       "version": "0.23.90",
       "port-version": 3
     },

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "935ceb14fbc62e378361d15466182ae4b88cf37a",
+      "git-tree": "78af6695c1ae1889782321149f71f12547f6ea66",
       "version": "0.23.90",
       "port-version": 3
     },

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "25b923000586733fcae49bf028e7d6be7aea5b1c",
+      "git-tree": "cbfdaff081849a8db3f594914f32408c9df4aec5",
       "version": "0.23.90",
       "port-version": 3
     },

--- a/versions/m-/mchehab-zbar.json
+++ b/versions/m-/mchehab-zbar.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "cbfdaff081849a8db3f594914f32408c9df4aec5",
+      "git-tree": "09371ba90d3c31d038bffd54a90203b5b4f7fdc8",
       "version": "0.23.90",
       "port-version": 3
     },

